compiler/rustc_passes/src/hir_id_validator.rs

```diff
@@ -8,34 +8,34 @@ use rustc_middle::hir::nested_filter;
 use rustc_middle::ty::TyCtxt;
 
 pub fn check_crate(tcx: TyCtxt<'_>) {
-    tcx.dep_graph.assert_ignored();
```

Comment:
Why this change?

Reply:
This is actually running inside of an eval-always query, not in `with_ignore()`. The two kinds both used `TaskDepsRef::Ignore` before this PR. Now that there is a `TaskDepsRef::EvalAlways`, the assertion fails.

Looking at the code again, I think we might actually want to just run it normally. I think the original `with_ignore()` scope was just a performance optimization when this check was run for release builds too.